### PR TITLE
#3157 Updates to integrate loader with self-hosted github runner

### PR DESCRIPTION
**Pull request by AndrewM-SDET**

### CODE REVIEW

File: integration/benchmark/cmd/ci/main.go [hunk 1]
```diff
@@ -281,3 +282,18 @@ func sendDataToBigQuery(
 	}
 	return nil
 }
+
+func outputLocalData(slices []dataSlice) {
+	jsonText, err := json.MarshalIndent(slices, "", "    ")
+	if err != nil {
+		println("Error converting slice data to json")
```
Comment (gomisha) [inline integration/benchmark/cmd/ci/main.go:289]: this is going to silently ignore the json marshaling error - can we use `log.Fatal()` instead?

File: integration/localnet/bootstrap.go [hunk 2]
```diff
@@ -310,7 +310,7 @@ func prepareServiceDirs(role string, nodeId string) (string, string) {
 	if nodeId != "" {
 		profilerDir = "./" + filepath.Join(ProfilerDir, role, nodeId)
 	}
-	err = os.MkdirAll(profilerDir, 0755)
+	err = os.MkdirAll(profilerDir, 0777)
```
Comment (gomisha) [inline integration/localnet/bootstrap.go:313]: can you add a comment on why we're giving full access to all users?

File: integration/localnet/bootstrap.go [hunk 3]
```diff
@@ -401,7 +401,7 @@ func prepareExecutionService(container testnet.ContainerConfig, i int, n int) Se
 
 	// create the execution state dir for the node
 	trieDir := "./" + filepath.Join(TrieDir, container.Role.String(), container.NodeID.String())
-	err := os.MkdirAll(trieDir, 0755)
+	err := os.MkdirAll(trieDir, 0777)
```
Comment (gomisha) [inline integration/localnet/bootstrap.go:404]: can you add a comment on why we're giving full access to all users?

File: integration/benchmark/cmd/ci/main.go [hunk 4]
```diff
@@ -281,3 +282,18 @@ func sendDataToBigQuery(
...
+	}
+
+	// output human-readable json blob
+	timestamp := time.Now()
+	fileName := fmt.Sprintf("tps-results-%v.json", timestamp.Format("2006-02-01 15:04"))
+	err = os.WriteFile(fileName, jsonText, 0666)
+	if err != nil {
+		fmt.Println(err)
```
Comment (gomisha) [inline integration/benchmark/cmd/ci/main.go:297]: this is going to silently ignore the json marshaling error - can we use log.Fatal() instead?

File: integration/localnet/Makefile [hunk 5]
```diff
@@ -70,6 +70,11 @@ start:
 	DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 docker-compose -f docker-compose.nodes.yml build
 	DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 docker-compose -f docker-compose.nodes.yml up -d
 
+.PHONY: start-thin
+start-thin:
```
Comment (gomisha) [inline integration/localnet/Makefile:74]: can you add a comment on what a "thin start" means and why we need it?

File: cmd/Dockerfile [hunk 6]
```diff
@@ -70,3 +70,5 @@ FROM gcr.io/distroless/base-debian11 as production-transit-nocgo
 COPY --from=build-transit-production-nocgo /app/app /bin/app
 
 ENTRYPOINT ["/bin/app"]
+
+FROM build-setup as environment-clean
```
Comment (AndrewM-SDET) [inline cmd/Dockerfile:74]: This thin docker image exists because I was unable to run a docker build on build-setup without major extra work, this thin image does all the same work but is available directly in the Makefile

File: integration/benchmark/cmd/ci/main.go (hunk 1 above)
Comment (AndrewM-SDET) [reply integration/benchmark/cmd/ci/main.go]: Updated

File: integration/localnet/bootstrap.go (hunk 2 above)
Comment (AndrewM-SDET) [reply integration/localnet/bootstrap.go]: This was an attempt at resolving this issue, reverted now that we have a more valid solution.

File: integration/benchmark/cmd/ci/main.go (hunk 4 above)
Comment (AndrewM-SDET) [reply integration/benchmark/cmd/ci/main.go]: Done

File: integration/localnet/Makefile (hunk 5 above)
Comment (AndrewM-SDET) [reply integration/localnet/Makefile:78]: Added, this resolves the port collision we were seeing on the runner.

**Comment by AndrewM-SDET** [comment]
bors retry

**Comment by AndrewM-SDET** [comment]
bors retry

File: integration/benchmark/cmd/ci/main.go [hunk 7]
```diff
@@ -103,9 +103,6 @@ func main() {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	sp := benchmark.NewStatsPusher(ctx, log, pushgateway, "loader", prometheus.DefaultGatherer)
```
Comment (pattyshack) [inline integration/benchmark/cmd/ci/main.go:106]: why was this deleted?  do you still need it for non-localDev mode?

**Review by pattyshack — APPROVED** [review]

File: integration/benchmark/cmd/ci/main.go (hunk 7 above)
Comment (AndrewM-SDET) [reply integration/benchmark/cmd/ci/main.go:113]: When running in CI we are not using prometheus and because of how the self-hosted runner works actually don't run the prometheus node at all (see `start-thin` in the Makefile). If we leave it in then the output is full of warnings instead of our data. This still exists in the manual version of the loader for when local testing is needed and prometheus is actually accessible.

Comment (SaveTheRbtz) [reply integration/benchmark/cmd/ci/main.go:113]: Do we have a Prometheus setup for this environment?  Can we just send the data there?

Comment (SaveTheRbtz) [reply integration/benchmark/cmd/ci/main.go:113]: (If there is no Prometheus setup there, you can consider hiding it behind the same `localDev` flag.)

File: integration/localnet/Makefile [hunk 8]
```diff
@@ -71,6 +73,13 @@ start:
 	DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 docker-compose -f docker-compose.nodes.yml build
```
Comment (SaveTheRbtz) [inline integration/localnet/Makefile:73]: you can consider using `$(MAKE) start-thin` here

Comment (SaveTheRbtz) [reply integration/localnet/Makefile:73]: (or possibly even renaming it so there is `start-with-metrics` and `start`.)

File: integration/benchmark/cmd/ci/main.go (hunk 7 above)
Comment (AndrewM-SDET) [reply integration/benchmark/cmd/ci/main.go:113]: There is no Prometheus on the CI environment, will hide this behind the localDev flag.

File: integration/localnet/Makefile (hunk 8 above)
Comment (AndrewM-SDET) [reply integration/localnet/Makefile:73]: I think we want to keep `start` as the main entrypoint but I can use `$(MAKE) start-thin` in start for reuse/clarity. Change incoming.

**Comment by AndrewM-SDET** [comment]
bors merge
